sympy/core/expr.py

```diff
@@ -299,11 +299,16 @@ def __ge__(self, other):
         n2 = _n2(self, other)
         if n2 is not None:
             return _sympify(n2 >= 0)
-        if self.is_extended_real or other.is_extended_real:
-            dif = self - other
-            if dif.is_extended_nonnegative is not None and \
-                    dif.is_extended_nonnegative is not dif.is_extended_negative:
-                return sympify(dif.is_extended_nonnegative)
+        if self.is_extended_real and other.is_extended_real:
+            if self.is_infinite and other.is_infinite:
+                if (self * other).is_extended_positive:  # inf < inf & -inf < -inf
+                    return S.true
```

Comment:
Although the comment says < it should really be >=. I think we can simplify this and the conditions below to
```
if (self.is_infinite and self.is_extended_positive) or (
    other.is_infinite and other.is_extended_negative):
    return S.True
```